### DISCUSSION TITLE
SQLPage and nginx in docker

### DISCUSSION THREAD

**Discussion by ionv06** [sqlpage/SQLPage#585] [Q&A] [answered]
Hi,

I configured three Docker containers - one for SQLPage, one for Nginx and one for mysql. Because I'm not very good at any of the three, I put the site files in the root folder, so that the url is of the form www.mysite.com (so no subfolder like www.mysite.com/ foo). Everything works, but I cannot configure nginx to add the '.sql' extension to the url when it is missing.
So, if the url is www.mysite.com/login.sql, everything works, if the url is www.mysite.com/login it doesn't work (I get 404 not found). 
I am attaching nginx configuration file; I start the nginx server with: docker run -it --rm --network my_network -d -p 80:80 -v $(pwd)/default.conf:/etc/nginx/conf.d/default.conf --name nginx web.
Thanks for any hint that might solve it!
[default.conf.txt](https://github.com/user-attachments/files/16990213/default.conf.txt)

**Comment by ionv06** [comment, marked as answer]
Solved with two rewrite

```nginx
    location / {
        rewrite ^(.+)/$ $1 permanent;
        rewrite ^(/[^.]*[^/])$ $1.sql break;
        proxy_pass http://sqlpage:8080;
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection 'upgrade';
        proxy_set_header Host $host;
        proxy_cache_bypass $http_upgrade;
    }
```

**Reply by lovasoa** [reply]
Great that you solved it ! Be careful though if you still want to use subfolders that contain an `index.sql`, though. This configuration looks like it will break the handling of `http://example.com/myfolder` when you have `/myfolder/index.sql` in your files. But as long as you don't use `index.sql`, this configuration looks fine

See:

https://sql.datapage.app/your-first-sql-website/nginx.sql

**Comment by lovasoa** [comment]
Hello ! 

  I just added a full example for using sqlpage together with nginx using docker to the repo: https://github.com/lovasoa/sqlpage/tree/main/examples/nginx

**Reply by ionv06** [reply]
Ophir, your solution is that of a professional,  I learned a lot from your example and it was very helpful, so thank you very much!
I made some minor changes in docker-compose.yml (file attached) to make it work on my configuration: I put container_name for each container and mounted sqlpage_socket and mysql_data from the current directory, after creating them beforehand. I had to do this because if docker creates them as volumes, sqlpage cannot write to the socket ('You do not have permission to bind to the UNIX socket "/tmp/sqlpage/sqlpage.sock". You can change the socket path in the configuration file or check the permissions.'). It's probably related to the rights of the current user, but I don't know how to manage the problem (I run docker on a raspberry pi 3 - just to see if it works ), so the only solution was to manually create the two folders (sqlpage_socket and mysql_data) and mount them. 
[docker-compose.yml.txt](https://github.com/user-attachments/files/17002635/docker-compose.yml.txt)
